packages/node_modules/@node-red/nodes/core/network/10-mqtt.js

```diff
@@ -996,12 +997,45 @@ module.exports = function(RED) {
         node.on('close', function(done) {
             node.disconnect(function() {
                 if(node.client) {
-                    node.client.removeAllListeners();
+                    node._clientRemoveListeners();
                 }
                 done();
             });
         });
 
+        // Helper functions to track the event listners we add to the
+        // client. The mqtt client also uses it own set of listeners
+        // so we can't use removeAllListeners() wothout breaking it
+
+        /**
+         * Add an event handlers to the MQTT.js client
+         * @param {string} [event] The name of the event (optional)
+         * @param {function} [handler] The handler for this event
+         */
+         node._clientOn = function(event, f) {
+            node.clientListeners.push({event, f})
+            node.client.on(event, f)
+        } 
```

Comment:
In this `_clientOn` function, both the event and handler are required but the JSDOC comment has them wrapped in `[]` - this makes them be reported in intellisense as optional. Also, you are pushing `{event, f}` to `clientListeners` but in `_clientRemoveListeners` you are checking for `event && event !== l.e` and  `handler && handler !== l.h`

### Before
![image](https://user-images.githubusercontent.com/44235289/167437112-4d4f0fcd-1f55-49cd-abdb-8f4da2bcc209.png)

### After
![image](https://user-images.githubusercontent.com/44235289/167437508-dca8e088-b792-491b-813c-b7d62f34e0a9.png)

<br>


I propose we use consistent naming throughout...

```suggestion
        /**
         * Add an event handler to the MQTT.js client
         * @param {string} event The name of the event
         * @param {function} handler The handler for this event
         */
         node._clientOn = function(event, handler) {
            node.clientListeners.push({event, handler})
            node.client.on(event, handler)
        } 
```